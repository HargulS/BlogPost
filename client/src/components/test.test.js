import React from 'react';
import PostJournalEntry from "./postJournalEntry";
import HomePage from "./homePage"
import HomepageListItems from "./homePageListItems"
import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })


  describe("rendering components",()=>{
    it('renders PostJournalEntry Correctly', () => {
      const wrapper = shallow(<PostJournalEntry/>)
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('renders Homepage Correctly', () => {
      const wrapper = shallow(<HomePage/>)
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('renders HomepageListItems Correctly', () => {
      const wrapper = shallow(<HomepageListItems/>)
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  })