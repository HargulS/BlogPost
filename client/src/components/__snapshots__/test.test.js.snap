// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components renders Homepage Correctly 1`] = `
<main
  className="homepage"
>
  <section
    className="homepage__linkBtnContainer"
  >
    <div
      className="homepage__btnContaier"
    >
      <button
        className="homepage__btn"
        onClick={[Function]}
      >
        Toggle By Date
      </button>
    </div>
    <Link
      className="homepage__link"
      to="/postdata"
    >
      Post Journal Entry
    </Link>
  </section>
  <section
    className="homepage__components"
  >
    <div
      className="homepage__mappedItems"
    />
    <div
      className="homepage__dataBydIdComponent"
    >
      <DataById />
    </div>
  </section>
</main>
`;

exports[`rendering components renders HomepageListItems Correctly 1`] = `
<main
  className="homepageList"
>
  <Link
    to="/undefined"
  >
    <section
      className="homepageList__listContainer"
    >
      <ul
        className="homepageList__listItems"
      >
        <li
          className="homepageList__listItemImgWrapper"
        >
          <img
            alt=""
            className="homepageList__listItemImg"
          />
        </li>
        <li
          className="homepageList__listItemTitle"
        >
          Title:
        </li>
        <li
          className="homepageList__listItemAuthor"
        >
          Name:
        </li>
        <li
          className="homepageList__listItemJournal"
        >
          Description:
        </li>
        <li
          className="homepageList__listItemDate"
        >
           Date:
        </li>
        <li
          className="homepageList__listItemReadMe"
        >
          Click To Read
        </li>
      </ul>
    </section>
  </Link>
</main>
`;

exports[`rendering components renders PostJournalEntry Correctly 1`] = `
<main
  className="post"
>
  <section
    className="post__formContaier"
  >
    <form
      className="post__form"
      onSubmit={[Function]}
    >
      <label>
        Title:
      </label>
      <textarea
        className="post__formTitle"
        onChange={[Function]}
        placeholder="Title"
        type="text"
      />
      <label>
        Journal Entry:
      </label>
      <textarea
        className="post__formEntry"
        onChange={[Function]}
        placeholder="Journal Entry"
        type="text"
      />
      <label
        className="post__labelUploadImg"
      >
        Upload Picture
      </label>
      <input
        className="post__inputUploadImg"
        name="myImage"
        onChange={[Function]}
        type="file"
      />
      <button
        className="form__button"
        onClick={[Function]}
      >
        Submit Entry
      </button>
    </form>
  </section>
  <section
    className="post__imageContainer"
  />
  <section
    className="post__linkContainer"
  >
    <Link
      className="post__link"
      to="/"
    >
      Back to HomePage
    </Link>
  </section>
</main>
`;
